### PR TITLE
`Adaptive learning`: Add legend to learning path graph representation

### LINKED ISSUE #7175: `Adaptive learning`: Add legend to learning path graph representation
### Is your feature request related to a problem?

_No response_

### Describe the solution you'd like

The graph representation should include a legend.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

To help users to better understand the learning path graph representation there should be a legend to better describe the meaning of the used icons.

Resolves #7175 

### Description

This PR adds a new component containing the legend.

### Steps for Testing

Prerequisites:
- 1 Students/Instructor
- 1 Course with learning paths enabled

1. Log in to Artemis
2. Navigate to a student's learning path and open the graph view
3. Make sure there is a legend in the bottom right corner of the graph
4. Make sure the translations are correctly displayed
5. Hover over the legend and check if the tooltips are displayed and correctly translated
6. Make sure the legend is readable in dark and light mode

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [learning-path-legend.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5505/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/learning-paths/learning-path-graph/learning-path-legend.component.ts.html) | 100% | ✅ |

�

### Screenshots

Screenshot 2023-09-28 at 15 16 10

Screenshot 2023-09-28 at 15 16 31

Screenshot 2023-09-28 at 15 16 47

Screenshot 2023-09-28 at 15 17 01

### CODE REVIEW

**Review by rriyaldhi — APPROVED** [review]
Tested on TS2 and works as described on the testing steps.

**Review by lennart-keller — APPROVED** [review]
Tested in testing session

**Review by pal03377 — DISMISSED** [review]
Tested on TS2 in testing session, legend looks good!

**Review by rstief — APPROVED** [review]
Tested on TS2

**Review by tobias-lippert — APPROVED** [review]
Legend is present and legible.

File: src/main/webapp/i18n/de/competency.json [hunk 1]
```diff
@@ -202,6 +197,39 @@
...
+                        "title": "Kompetenzbeginn",
+                        "tooltip": "Hier beginnt eine Kompetenz. Die Knoten zwischen Kompetenzbeginn und -ende sind Inhalte die dir beim erreichen dieser Kompetenz helfen."
+                    },
+                    "competencyEnd": {
+                        "title": "Kompetenzende",
+                        "tooltip": "Hier endet eine Kompetenz. Die Knoten zwischen Kompetenzbeginn und -ende sind Inhalte die dir beim erreichen dieser Kompetenz helfen."
+                    },
+                    "matchStart": {
+                        "title": "Verzweigung",
+                        "tooltip": "Hier hast du die Wahl an welcher der folgenden Kompetenzen du arbeiten möchtest. Die Kompetenzen wurden von deinem Instruktor als gleichbedeutend hinterlegt, d.h. eine der Kompetenzen zu meistern ist hinreichend."
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/de/competency.json:218]: Extract from our language guide:
> Terms replaced in the Artemis user interface
Dozent, Instruktor → Lehrende (Plural), Lehrkraft (Singular)

**Review by tobias-lippert — APPROVED** [review]
Code LGTM

**Review by tobias-lippert — APPROVED** [review]
Reapprove

**Review by rstief — APPROVED** [review]
Reapproved after translation fix

**Review by JohannesStoehr — APPROVED** [review]
Language is now in accordance with our guidelines

**Review by lennart-keller — DISMISSED** [review]
Reapprove after fixing translation

**Review by rriyaldhi — APPROVED** [review]
Reapproved after translation fix.

**Review by JohannesStoehr — APPROVED** [review]
Reapprove code after merge commits and base branch change

**Review by tobias-lippert — APPROVED** [review]
Reapprove after base branch change

**Review by lennart-keller — APPROVED** [review]
Reapprove after base branch change

**Review by rstief — APPROVED** [review]

**Review by pal03377 — APPROVED** [review]

**Pull request merged**